### PR TITLE
[FIX] ADNI-to-BIDS : incorporate the fix for KeyError "APGEN"

### LINKED ISSUE #1294: Key Error : APGEN1
Hello, 

I used Clinica to convert ADNI to BIDS and I have a pandas error : "KeyError : APGEN1" 

This error concern adni_to_bids.convert_clinical_data with participants_df = bids.create_participants_df() and more precisely value_to_read = file_to_read[participant_field_db[i]]

It's when participants.tsv are created and no tsv are generated in my case. I just download all ADNI data between ADNI 1 to ADNI 4 (on July 2024) et just run clinica without options (for all datas). 

Do you have an idea why I have this error ? I suppose that one key isn't here in clinical data for ADNI but if we have a larger problem, I would like to submit it. 

Thank you for your help !

### PR DESCRIPTION
**Pull request by AliceJoubert**
Closes #1294

### CODE REVIEW

File: clinica/iotools/bids_utils.py [hunk 1]
```diff
@@ -401,6 +401,17 @@ def create_participants_df(
                     file_to_read = load_clinical_csv(
                         clinical_data_dir, location.split(".")[0]
                     )
+                    if location == "APOERES.csv":
```
Comment (NicolasGensollen) [inline clinica/iotools/bids_utils.py:404]: This is only for adni-to-bids right ? Maybe add something like:

```python
if study_name == StudyName.ADNI and location == "APOERES.csv":
    ...
```

File: clinica/iotools/bids_utils.py [hunk 2]
```diff
@@ -401,6 +401,17 @@ def create_participants_df(
...
+                        if (
```
Comment (NicolasGensollen) [inline clinica/iotools/bids_utils.py:405]: Maybe add a small comment with the link to the issue

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @AliceJoubert !

File: clinica/iotools/bids_utils.py (hunk 1 above)
Comment (AliceJoubert) [reply clinica/iotools/bids_utils.py:404]: Yes this is for ADNI only ; the file is specific to ADNI too

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @AliceJoubert !

**Pull request merged**